subsys/net/lib/config/Kconfig

```diff
@@ -234,6 +234,13 @@ config NET_CONFIG_SNTP_INIT_SERVER
 	  e.g. server information at
 	  https://support.ntp.org/bin/view/Servers/NTPPoolServers
 
+config NET_CONFIG_SNTP_INIT_SERVER_USE_DHCPV4_OPTION
+	bool "SNTP server to use for system clock init is set using DHCPv4 option"
+	depends on NET_DHCPV4_OPTION_NTP_SERVER
```

Comment:
I think it makes sense to enable this by default if the user has enabled the NTP server option so I suggest `default y` here.